plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -565,6 +574,15 @@ private static Document orPredicate(List<Document> values)
         return new Document(OR_OP, values);
     }
 
+    private static Document andPredicate(List<Document> values)
+    {
+        checkState(!values.isEmpty());
+        if (values.size() == 1) {
+            return values.get(0);
+        }
+        return new Document(AND_OP, values);
```

Comment:
Doing this explicit AND and is to ensure that we won't override the filter for a specific column right ?

Reply:
Or will `Document#putAll` handles it internally ?

Reply:
> Doing this explicit AND and is to ensure that we won't override the filter for a specific column right ?

Right. Unfortunately, `Document#putAll` overwrites existing entries in case of duplicated keys.

Reply:
Can we add this an a comment ?